docs/src/cookbooks/hierarchy.md

```diff
@@ -161,8 +166,10 @@ class Top extends Module {
   println(s"Width is: ${definition.width}")
 }
 ```
-```scala mdoc:verilog
-chisel3.stage.ChiselStage.emitVerilog(new Top())
+
+<!-- TODO: This currently does not work. -->
+```scala
+circt.stage.ChiselStage.emitSystemVerilog(new Top())
```

Comment:
This test errors in CIRCT with a StageError and little additional information. I'm actually very confused by this because the above `AddOne` doesn't even appear to be legal Scala. It's doing something like:

```scala
class Foo(val a: Int) {
  val a = a
}
```

Reply:
what do you mean? It's doing out := in + 1

?

Oh, you mean the val width = width? The compiler plugin should be doing things with the @public that make that legal (yes, it's weird). I'm not sure how that would be different for circt stage though?

Reply:
It seems really concerning to me that we're rewriting illegal Scala code...

Can this instead be written as:

```scala
class Foo(_a: Int) {
  val a = _a
}
```

Can the `@public` rewriter cause _new_ name collisions with its rewriting?

There's the secondary problem in the above where `val a = a`, ignoring the double definition, ambiguously typed as it appears recursive.

Reply:
I'll dig into what's going on with CIRCT here. Part of what was flummoxing me was that I couldn't run this snippet in `Scala-cli` because I was forgot to add whatever the compiler plugin is that does the D/I magic.

Reply:
It's not compiler plugin, it's just the `@instantiable` macro. Doesn't really make a difference though, it's very weird that this doesn't work with CIRCT as there should be no difference...

Reply:
The issue is an abstract reset in the definition module which is then included in the compile.

Jack pointed me at `VerilogMdocModifier.scala` which needs to be fixed up before this PR lands.